src/data/DataElementsD2Repository.ts

```diff
@@ -1,23 +1,31 @@
 import _ from "lodash";
-import { D2Api, Id } from "types/d2-api";
+import { D2Api, Id, MetadataPick } from "types/d2-api";
 import { NamedRef } from "domain/entities/Base";
 import { DataElementsRepository } from "domain/repositories/DataElementsRepository";
+import { DataElement } from "domain/entities/DataElement";
 
 export class DataElementsD2Repository implements DataElementsRepository {
     constructor(private api: D2Api) {}
 
+    async getByIds(ids: Id[]): Promise<DataElement[]> {
+        return this.getDataElements(ids);
+    }
+
     async getDataElementsNames(ids: Id[]): Promise<NamedRef[]> {
```

Comment:
`getByIds` is returning `DataElement[]`, which makes a lot more sense that `getDataElementsNames` returning `NamedRef[]`. And, since DataElement is compatible with NamedRef, I'd remove getDataElementsNames and make everybody use getByIds.